File: src/app/models/comic-edition.ts
/*
Required Notice: Copyright (C) 2026 José Elias Hermínio Mendes - Projeto Comic Organizer

Este software é licenciado sob a PolyForm Noncommercial License 1.0.0.
O uso, modificação e distribuição são permitidos apenas para fins NÃO COMERCIAIS.
Para ler a licença completa, veja o arquivo LICENSE.txt no diretório raiz.
*/

import { ComicPage } from "./comic-page";

// Agrupa as paginas e metadados de uma edicao carregada no app.
export interface ComicEdition {
  id: number;
  title: string;
  pages: ComicPage[];

  selected?: boolean;
  expanded?: boolean;

  originalFile?: File;

  converted?: boolean;

  outputPath?: string;
}

File: src-tauri/src/models/comic_edition.rs
/*
Required Notice: Copyright (C) 2026 José Elias Hermínio Mendes - Projeto Comic Organizer

Este software é licenciado sob a PolyForm Noncommercial License 1.0.0.
O uso, modificação e distribuição são permitidos apenas para fins NÃO COMERCIAIS.
Para ler a licença completa, veja o arquivo LICENSE.txt no diretório raiz.
*/

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ComicPage {
    pub id: usize,
    pub file_name: String,
    pub image_path: String,
    pub page_number: usize,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ComicEdition {
    pub id: usize,
    pub title: String,
    pub pages: Vec<ComicPage>,
}
